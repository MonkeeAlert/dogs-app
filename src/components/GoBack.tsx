import {View, Text, Pressable, StyleSheet} from 'react-native';
import React from 'react';
import {Icon} from 'react-native-elements';
import {colors} from '../utils/constants';
import {useNavigation} from '@react-navigation/native';

const GoBack = () => {
  const {goBack} = useNavigation();

  return (
    <View style={styles.container}>
      <Pressable onPress={goBack}>
        <Icon
          name={'arrow-back'}
          type={'ionicon'}
          tvParallaxProperties={false}
          size={18}
          color={colors.darkGray}
        />
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 14,
    left: 14,
    zIndex: 1,
    elevation: 1,
    backgroundColor: colors.white,
    width: 35,
    height: 35,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 35,
  },
});

export default GoBack;
