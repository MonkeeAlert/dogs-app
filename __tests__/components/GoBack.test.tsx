import React from 'react';
import renderer from 'react-test-renderer';
import GoBack from '../../src/components/GoBack';
import {shallow} from 'enzyme';

describe('GoBack', () => {
  it('snapshot', () => {
    const tree = renderer.create(<GoBack />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('event', () => {
    const wrapper = shallow(<GoBack />)
      .find({testID: 'GoBack_Button'})
      .first();
    expect(wrapper.exists()).toBe(true);

    const wrapperWithProps = wrapper.props();

    const handlerKeyId = Object.keys(wrapperWithProps).indexOf('onPress');
    expect(handlerKeyId !== -1).toBe(true);

    const handlerValueId = typeof Object.values(wrapperWithProps)[handlerKeyId];
    expect(handlerValueId !== 'undefined').toBe(true);
  });
});
