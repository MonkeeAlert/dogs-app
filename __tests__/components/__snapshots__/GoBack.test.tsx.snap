// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBack snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fffafb",
      "borderRadius": 35,
      "elevation": 1,
      "height": 35,
      "justifyContent": "center",
      "left": 14,
      "position": "absolute",
      "top": 14,
      "width": 35,
      "zIndex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="GoBack_Button"
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#2b2c2c",
                  "fontSize": 18,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            ï„¤
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
